File: clients/go/connect_test.go
package afhe

import (
	"context"
	"encoding/json"
	"net/http"
	"net/http/httptest"
	"reflect"
	"testing"
)

func TestConnectAutoLoadsDefaultKeys(t *testing.T) {
	t.Helper()

	var paths []string
	var loaded LoadOptions

	srv := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		paths = append(paths, r.URL.Path)

		switch r.URL.Path {
		case "/health":
			_ = json.NewEncoder(w).Encode(map[string]string{"status": "ok"})
		case "/load":
			if err := json.NewDecoder(r.Body).Decode(&loaded); err != nil {
				t.Fatalf("decode load body: %v", err)
			}
			_ = json.NewEncoder(w).Encode(map[string][]string{"loaded": {"skb", "pkb", "dictb"}})
		default:
			http.NotFound(w, r)
		}
	}))
	defer srv.Close()

	client, err := Connect(context.Background(), ConnectOptions{
		BaseURL:     srv.URL,
		InsecureTLS: Bool(true),
	})
	if err != nil {
		t.Fatalf("Connect: %v", err)
	}
	if client == nil {
		t.Fatal("Connect returned nil client")
	}

	if !reflect.DeepEqual(paths, []string{"/health", "/load"}) {
		t.Fatalf("unexpected request order: %#v", paths)
	}
	if loaded != (LoadOptions{SKB: "file/skb", PKB: "file/pkb", DictB: "file/dictb"}) {
		t.Fatalf("unexpected load options: %#v", loaded)
	}
}

func TestConnectRejectsRemoteInsecureTLS(t *testing.T) {
	t.Helper()

	_, err := Connect(context.Background(), ConnectOptions{
		BaseURL:     "https://api.example.com:8443",
		InsecureTLS: Bool(true),
		HealthCheck: Bool(false),
		AutoLoad:    Bool(false),
	})
	if err == nil {
		t.Fatal("expected insecure remote TLS to be rejected")
	}
}

File: clients/typescript/src/connect.test.ts
import { connect } from "./connect";

describe("connect()", () => {
  test("health-checks and auto-loads default keys", async () => {
    const fetchMock = jest
      .fn()
      .mockResolvedValueOnce(new Response(JSON.stringify({ status: "ok" }), { status: 200 }))
      .mockResolvedValueOnce(
        new Response(JSON.stringify({ loaded: ["skb", "pkb", "dictb"] }), { status: 200 }),
      );

    const client = await connect({
      baseUrl: "https://localhost:8443",
      fetch: fetchMock as unknown as typeof fetch,
    });

    expect(client).toBeDefined();
    expect(fetchMock).toHaveBeenCalledTimes(2);
    expect(fetchMock.mock.calls[0][0]).toBe("https://localhost:8443/health");
    expect(fetchMock.mock.calls[1][0]).toBe("https://localhost:8443/load");
    expect(JSON.parse(String(fetchMock.mock.calls[1][1]?.body))).toEqual({
      skb: "file/skb",
      pkb: "file/pkb",
      dictb: "file/dictb",
    });
  });

  test("rejects insecure TLS outside localhost", async () => {
    await expect(
      connect({
        baseUrl: "https://api.example.com:8443",
        insecureTLS: true,
      }),
    ).rejects.toThrow("insecureTLS is only allowed for localhost");
  });
});
